lib/charms/opensearch/v0/opensearch_plugins.py

```diff
@@ -332,6 +332,41 @@ class PluginState(BaseStrEnum):
     WAITING_FOR_UPGRADE = "waiting-for-upgrade"
 
 
+class OpenSearchPluginRelationsHandler(ABC):
+    """Implements the relation manager for each plugin.
+
+    Plugins may have one or more relations tied to them. This abstract class
+    enables different modules to implement a class that can specify which
+    relations should plugin manager listen to.
+    """
+
+    _singleton = None
```

Comment:
can you explain why?

Reply:
Because we cannot have two or more objects created in the same hook, pointing to the same relation. Having a singleton makes this class more flexible, so the plugin manager can be called several times, each time it either allocates or reuses the same singleton, when needed. Otherwise the plugin manager would have to allocate this handler for each plugin case.